File: plugins/togetherai-skills/skills/together-chat-completions/scripts/chat_basic.py
#!/usr/bin/env python3
"""
Together AI Chat Completions — Basic Chat and Streaming (v2 SDK)

Demonstrates single-query chat, streaming, and multi-turn conversation.

Usage:
    python chat_basic.py

Requires:
    uv pip install "together>=2.0.0"
    export TOGETHER_API_KEY=your_key
"""

from together import Together

client = Together()


def basic_chat() -> None:
    """Send a single chat completion request."""
    print("=== Basic Chat ===")
    response = client.chat.completions.create(
        model="openai/gpt-oss-20b",
        messages=[{"role": "user", "content": "What are some fun things to do in NYC?"}],
    )
    print(response.choices[0].message.content)
    print()


def streaming_chat() -> None:
    """Stream tokens incrementally."""
    print("=== Streaming ===")
    stream = client.chat.completions.create(
        model="openai/gpt-oss-20b",
        messages=[{"role": "user", "content": "Write a haiku about coding"}],
        stream=True,
    )
    for chunk in stream:
        if chunk.choices:
            print(chunk.choices[0].delta.content or "", end="", flush=True)
    print("\n")


def multi_turn_chat() -> None:
    """Multi-turn conversation with system prompt."""
    print("=== Multi-Turn ===")
    messages = [
        {"role": "system", "content": "You are a helpful travel guide. Keep answers brief."},
        {"role": "user", "content": "What should I do in Paris?"},
    ]

    response = client.chat.completions.create(
        model="openai/gpt-oss-20b",
        messages=messages,
    )
    assistant_reply = response.choices[0].message.content
    print(f"User: What should I do in Paris?")
    print(f"Assistant: {assistant_reply}\n")

    # Continue the conversation
    messages.append({"role": "assistant", "content": assistant_reply})
    messages.append({"role": "user", "content": "How about food recommendations?"})

    response = client.chat.completions.create(
        model="openai/gpt-oss-20b",
        messages=messages,
    )
    print(f"User: How about food recommendations?")
    print(f"Assistant: {response.choices[0].message.content}")


if __name__ == "__main__":
    basic_chat()
    streaming_chat()
    multi_turn_chat()

File: plugins/togetherai-skills/skills/together-chat-completions/scripts/chat_basic.ts
#!/usr/bin/env -S npx tsx
/**
 * Together AI Chat Completions — Basic Chat and Streaming
 *
 * Demonstrates single-query chat, streaming, and multi-turn conversation.
 *
 * Usage:
 *   npx tsx chat_basic.ts
 *
 * Requires:
 *   npm install together-ai
 *   export TOGETHER_API_KEY=your_key
 */

import Together from "together-ai";

const client = new Together({
  apiKey: process.env.TOGETHER_API_KEY,
});

async function basicChat(): Promise<void> {
  console.log("=== Basic Chat ===");
  const response = await client.chat.completions.create({
    model: "openai/gpt-oss-20b",
    messages: [{ role: "user", content: "What are some fun things to do in NYC?" }],
  });
  console.log(response.choices[0].message.content);
  console.log();
}

async function streamingChat(): Promise<void> {
  console.log("=== Streaming ===");
  const stream = await client.chat.completions.create({
    model: "openai/gpt-oss-20b",
    messages: [{ role: "user", content: "Write a haiku about coding" }],
    stream: true,
  });

  for await (const chunk of stream) {
    process.stdout.write(chunk.choices[0]?.delta?.content || "");
  }
  console.log("\n");
}

async function multiTurnChat(): Promise<void> {
  console.log("=== Multi-Turn ===");
  const messages: { role: string; content: string }[] = [
    { role: "system", content: "You are a helpful travel guide. Keep answers brief." },
    { role: "user", content: "What should I do in Paris?" },
  ];

  const response = await client.chat.completions.create({
    model: "openai/gpt-oss-20b",
    messages: messages as any,
  });
  const assistantReply = response.choices[0].message.content ?? "";
  console.log("User: What should I do in Paris?");
  console.log(`Assistant: ${assistantReply}\n`);

  messages.push({ role: "assistant", content: assistantReply });
  messages.push({ role: "user", content: "How about food recommendations?" });

  const response2 = await client.chat.completions.create({
    model: "openai/gpt-oss-20b",
    messages: messages as any,
  });
  console.log("User: How about food recommendations?");
  console.log(`Assistant: ${response2.choices[0].message.content}`);
}

async function main(): Promise<void> {
  await basicChat();
  await streamingChat();
  await multiTurnChat();
}

main();
